nifi-nar-bundles/nifi-druid-bundle/nifi-druid-controller-service/src/main/java/org/apache/nifi/controller/DruidTranquilityController.java

```diff
@@ -0,0 +1,416 @@
+		while(aggIterator.hasNext()){
+			currFactory = aggIterator.next();
+			getLogger().debug("Verifying Aggregator Definition");
+			getLogger().debug("Aggregator Name: " + currFactory.getName());
+			getLogger().debug("Aggregator Type: " + currFactory.getTypeName());
+			getLogger().debug("Aggregator Req Fields: " + currFactory.requiredFields());			
+		}
+		// Tranquility uses ZooKeeper (through Curator) for coordination.
+		final CuratorFramework curator = CuratorFrameworkFactory
+		                .builder()
+		                .connectString(zkConnectString)
+		                .retryPolicy(new ExponentialBackoffRetry(1000, 20, 30000))
+		                .build();
+		curator.start();
+
+		// The JSON serialization of your object must have a timestamp field in a format that Druid understands. By default,
+		// Druid expects the field to be called "timestamp" and to be an ISO8601 timestamp.
+		final TimestampSpec timestampSpec = new TimestampSpec(timestampField, "auto", null);
+		        
+		final Beam<Map<String, Object>> beam = DruidBeams.builder(timestamper)
+			    		.curator(curator)
+		                .discoveryPath(discoveryPath)
+		                .location(DruidLocation.create(DruidEnvironment.create(indexService, firehosePattern),dataSource))
+		                .timestampSpec(timestampSpec)
+		                .rollup(DruidRollup.create(DruidDimensions.specific(dimensions), aggregator, QueryGranularity.fromString(queryGranularity)))
+		                .tuning(
+		                        ClusteredBeamTuning
+		                                .builder()
+		                                //.segmentGranularity(Granularity.MINUTE)
+		                                .segmentGranularity(getSegmentGranularity(segmentGranularity))
+		                                .windowPeriod(new Period(windowPeriod))
+		                                .partitions(clusterPartitions)
+		                                .replicants(clusterReplication)
+		                                .build()
+		                )
+		                .druidBeamConfig(
+		                        DruidBeamConfig
+		                                .builder()
+		                                .indexRetryPeriod(new Period(indexRetryPeriod))
+		                                .build())
+		                .buildBeam();
+		
+		tranquilizer = Tranquilizer.builder()
+                .maxBatchSize(10000000)
+                .maxPendingBatches(1000)
+                .lingerMillis(1000)
+                .blockOnFull(true)
+                .build(beam);
+		
+        tranquilizer.start();
+	}
+	
+	public Tranquilizer getTranquilizer(){
+		return tranquilizer;
+	}
+	
+	private  List<Map<String, String>> parseJsonString(String aggregatorJson) {
+        ObjectMapper mapper = new ObjectMapper();
+        List<Map<String, String>> aggSpecList = null;
+        try {
+        	getLogger().debug("Druid Tranquility Service: Aggregator Spec as String: " + aggregatorJson);
```

Comment:
id recommend getting rid of all these debugs or guarding them somehow or using the parameter based strings so that we're not creating the concatenated string object ahead of time.  Ie instead of "string"+"otherstring" use logger.debug("Druid stuff {}",new Object[]{aggregatorJson}))

Reply:
Removed debug lines. Will be reflected in the update.

Reply:
Removed debug lines. Will be reflected in the update.